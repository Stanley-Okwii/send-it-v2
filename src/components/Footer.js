import React from 'react'
import FilterLink from '../containers/FilterLink'
import ToggleTab from '../containers/ToggleTab';
import { VisibilityFilters } from '../actions'

const Footer = () => (
  <div>
    <span>Show: </span>
    <FilterLink filter={VisibilityFilters.SHOW_ALL}>All</FilterLink>
    <FilterLink filter={VisibilityFilters.SHOW_ACTIVE}>Active</FilterLink>
    <FilterLink filter={VisibilityFilters.SHOW_COMPLETED}>Completed</FilterLink>
    <ToggleTab/>
  </div>
)

export default Footer
