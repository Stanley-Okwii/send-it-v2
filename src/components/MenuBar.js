import React, { Component } from 'react'
import { Menu } from 'semantic-ui-react'

export class MenuBar extends Component {
  state = { activeItem: 'home' }

  handleItemClick = (e, { name }) => this.setState({ activeItem: name })

  render() {
    const { activeItem } = this.state

    return (
      <Menu secondary>
        <Menu.Item name='home' active={activeItem === 'home'} onClick={
          this.handleItemClick
          } />
        <Menu.Item
          name='profile'
          active={activeItem === 'profile'}
          onClick={this.handleItemClick}
        />
        <Menu.Menu position='right'>
          <Menu.Item
            name='logout'
            active={activeItem === 'logout'}
            onClick={this.handleItemClick}
          />
        </Menu.Menu>
      </Menu>
    )
  }
}

export default MenuBar;