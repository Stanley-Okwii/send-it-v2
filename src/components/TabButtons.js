import React from 'react'
import { Button } from 'semantic-ui-react';
import PropTypes from 'prop-types'


const TabButtons = ({ onClickLogIn, onClickSignUp }) => (
    <Button.Group>
        <Button icon='sign-in' onClick={() => onClickLogIn()}
            content='Sign In' />
        <Button icon='user plus' onClick={() => onClickSignUp()
        }
            content='Sign Up' />
    </Button.Group>
)

TabButtons.propTypes = {
    onClickSignUp: PropTypes.func.isRequired,
    onClickLogIn: PropTypes.func.isRequired
}

export default TabButtons
