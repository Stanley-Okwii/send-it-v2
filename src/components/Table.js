import React, { Component} from "react";
import { Table } from "semantic-ui-react";
import TableContainer from "../containers/TableContainer";

export class ParcelTable extends Component {
  render() {
    return (
      <Table compact celled definition>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Date</Table.HeaderCell>
            <Table.HeaderCell>Parcel</Table.HeaderCell>
            <Table.HeaderCell>Parcel ID</Table.HeaderCell>
            <Table.HeaderCell>Weight(Kgs)</Table.HeaderCell>
            <Table.HeaderCell>Price(Ugx)</Table.HeaderCell>
            <Table.HeaderCell>Recepient</Table.HeaderCell>
            <Table.HeaderCell>Pick up</Table.HeaderCell>
            <Table.HeaderCell>Destination</Table.HeaderCell>
            <Table.HeaderCell>Current location</Table.HeaderCell>
            <Table.HeaderCell>Status</Table.HeaderCell>
          </Table.Row>
        </Table.Header>

        <TableContainer />

        <Table.Footer fullWidth>
          <Table.Row />
        </Table.Footer>
      </Table>
    );
  }
}

export default ParcelTable;
