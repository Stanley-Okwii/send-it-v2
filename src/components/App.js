import React, { Component } from 'react'
import Footer from './Footer'
import AddTodo from '../containers/AddTodo'
import VisibleTodoList from '../containers/VisibleTodoList'

 export class App extends Component {

  render() {

    return (
      <div>
        <AddTodo />
        <VisibleTodoList />
        <Footer />
      </div>
    );
  }

  componentWillUnmount(){
    console.log("component will unmount.....");
  }

  componentWillMount(){
    console.log("component will mount.....");
  }

  componentDidMount(){
    console.log("component did mount.....");
  }
}

// export default App
//