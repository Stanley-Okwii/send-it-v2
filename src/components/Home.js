import React, { Component } from "react";
import { Grid} from "semantic-ui-react";

import { SemanticToastContainer} from "react-semantic-toasts";
import "react-semantic-toasts/styles/react-semantic-alert.css";
import "semantic-ui-css/semantic.min.css";
import "../css/App.css";

export class Home extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLogInTab: true,
    };
    let { history } = this.props;
    history.push({
      pathname: '/home'
     });
  }

  render() {
    return (
      <div>
        <SemanticToastContainer position="top-center" />
        <Grid
          textAlign="center"
          style={{ height: "60%" }}
          verticalAlign="middle"
        >
        Home
        </Grid>
      </div>
    );
  }
}
export default Home;
