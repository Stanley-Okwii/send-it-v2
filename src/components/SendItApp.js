import React, { Component } from "react";
import { withRouter } from 'react-router-dom';
import store from "../store";
import { Form, Grid, Header, Button, Segment, Image } from "semantic-ui-react";

import { SemanticToastContainer, toast } from "react-semantic-toasts";
import "react-semantic-toasts/styles/react-semantic-alert.css";
import "semantic-ui-css/semantic.min.css";
import "../css/App.css";

export class SendItApp extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLogInTab: true,
      userSignUpData: {
        username: "",
        email: "",
        password1: "",
        password2: ""
      }
    };
    let { history } = this.props;
    history.push({
      pathname: '/'
     });
  }

  render() {
    const isLogInTab = store.getState().isLogInTab;
    console.log(isLogInTab);
    return (
      <div>
        <SemanticToastContainer position="top-center" />
        <Grid
          textAlign="center"
          style={{ height: "60%" }}
          verticalAlign="middle"
        >
          <Grid.Column style={{ maxWidth: 600 }}>
            <Header as="h2" color="teal" textAlign="center">
              SendIT Courier Services
              <Image src="logo.png" />
            </Header>
            <Form className="form-wrapper" size="large">
              <Button.Group>
                <Button
                  icon="sign-in"
                  onClick={() => this.showLogin()}
                  content="Sign In"
                />
                <Button
                  icon="user plus"
                  onClick={() => this.showSignUp()}
                  content="Sign Up"
                />
              </Button.Group>
              {this.state.isLogInTab ? (
                <Segment vertical>
                  <Form.Field>
                    <label className="label-align">Email</label>
                    <input placeholder="Email" />
                  </Form.Field>
                  <Form.Field>
                    <label className="label-align">Password</label>
                    <input type="password" placeholder="Password" />
                  </Form.Field>
                  <Button className="index-button" color="teal">
                    Login
                  </Button>
                </Segment>
              ) : (
                <Segment vertical>
                  <Form.Field>
                    {/* <label className="label-align">User Name</label>
                    <input
                      placeholder="User Name"
                      onChange={(event = this.handleEmailChange(event))}
                    /> */}
                    <Form.Input
                      name="username"
                      placeholder="username"
                      value={this.state.userSignUpData.username}
                      onChange={this.onTextChange}
                    />
                  </Form.Field>
                  <Form.Field>
                    {/* <label className="label-align">Email</label>
                    <input type="email" placeholder="Email" /> */}
                    <Form.Input
                      name="email"
                      placeholder="email"
                      value={this.state.userSignUpData.email}
                      onChange={this.onTextChange}
                    />
                  </Form.Field>
                  <Form.Field>
                    <Form.Input
                      name="password1"
                      placeholder="password"
                      type="password"
                      value={this.state.userSignUpData.password1}
                      onChange={this.onTextChange}
                    />
                  </Form.Field>
                  <Form.Field>
                    {/* <label className="label-align">Confirm password</label>
                    <input type="password" placeholder="Confirm password" /> */}
                    <Form.Input
                      name="password2"
                      type="password"
                      placeholder="confirm password"
                      value={this.state.userSignUpData.password2}
                      onChange={this.onTextChange}
                    />
                  </Form.Field>
                  <Button onClick={this.registerUser} className="index-button" color="teal">
                    sign up
                  </Button>
                </Segment>
              )}
            </Form>
          </Grid.Column>
        </Grid>
      </div>
    );
  }

  showLogin = () => {
    this.setState({ isLogInTab: true });
  };

  showSignUp = () => {
    this.setState({ isLogInTab: false });
  };

  onTextChange = (_event, { name, value }) => {
    if (name) {
      if (name === "username") {
        this.setState({
          userSignUpData: { ...this.state.userSignUpData, username: value }
        });
      } else if (name === "email") {
        this.setState({
          userSignUpData: { ...this.state.userSignUpData, email: value }
        });
      } else if (name === "password1") {
        this.setState({
          userSignUpData: { ...this.state.userSignUpData, password1: value }
        });
      } else if (name === "password2") {
        this.setState({
          userSignUpData: { ...this.state.userSignUpData, password2: value }
        });
      }
    }
  };

  registerUser = () => {
    // const url = "https://sender-app.herokuapp.com/api/v1/user";
    const url = 'http://127.0.0.1:5000/api/v1/user'
    const { email, password1, username} = this.state.userSignUpData
    const raw_data = {
      "email": email,
      "password": password1,
      "name": username
    }

    const request = {
      method: "POST",
      body: JSON.stringify(raw_data),
      headers: {
        "Content-Type": "application/json"
      }
    };
    fetch(url, request)
      .then(response => response.json())
      .then(data => {
        const message = data["message"];
        if (message === "successfully created a new account") {
          toast(
            {
              title: "Sign up",
              type: "success",
              description: <p>Successfully created an account</p>
            },
            () => console.log("toast closed"),
            () => console.log("toast clicked")
          );
          this.props.history.push('/app')
          console.log(this.props.history);
        } else {
          toast(
            {
              title: "Failed to sign up",
              type: "error",
              description: <p>{message}</p>
            },
            () => console.log("toast closed"),
            () => console.log("toast clicked")
          );
          console.log("failed login");
        }
      })
      .catch(function(error) {
        console.log("Error: " + error);
      });
  };
}
export default withRouter(SendItApp);
