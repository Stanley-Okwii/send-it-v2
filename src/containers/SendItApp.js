import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { Form, Grid, Header, Button, Image } from "semantic-ui-react";

import LoginContaner from "../../src/containers/LoginContainer";
import SignUpContainer from "../../src/containers/SignUpContainer";

import { toggleTab } from "../actions/registration/login";
import "semantic-ui-css/semantic.min.css";

import "../css/App.css";

export class SendItApp extends Component {
  constructor(props) {
    super(props);
    let { history } = this.props;
    history.push({
      pathname: "/"
    });
  }

  render() {
    return (
      <div>
        <Grid
          textAlign="center"
          style={{ height: "60%" }}
          verticalAlign="middle"
        >
          <Grid.Column style={{ maxWidth: 600 }}>
            <Header as="h2" color="teal" textAlign="center">
              SendIT Courier Services
              <Image src="logo.png" />
            </Header>
            <Form className="form-wrapper" size="large">
              <Button.Group>
                <Button
                  icon="sign-in"
                  onClick={() => this.props.toggleTab(true)}
                  content="Sign In"
                />
                <Button
                  id="signup"
                  icon="user plus"
                  onClick={() => this.props.toggleTab(false)}
                  content="Sign Up"
                />
              </Button.Group>
              {this.props.isLogInTab ? <LoginContaner /> : <SignUpContainer />}
            </Form>
          </Grid.Column>
        </Grid>
      </div>
    );
  }
}


export const mapStateToProps = state => {
  return {
    ...state,
    isLogInTab: state.isLogInTab
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    toggleTab: function(value) {
      dispatch(toggleTab(value));
    }
  };
};

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(SendItApp));
