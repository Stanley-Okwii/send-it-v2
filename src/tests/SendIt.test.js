import React from "react";
import { shallow } from "enzyme";
import { SendItApp } from '../components/SendItApp';

it('renders without crashing', () => {
    const wrapper = shallow(<SendItApp
        history = {{ push: jest.fn()}}
    />);
    expect(wrapper).toMatchSnapshot();
});
