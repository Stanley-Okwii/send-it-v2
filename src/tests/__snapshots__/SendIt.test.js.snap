// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SendItApp
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "pathname": "/",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SemanticToastContainer
          animation={null}
          className=""
          position="top-center"
        />,
        <Grid
          style={
            Object {
              "height": "60%",
            }
          }
          textAlign="center"
          verticalAlign="middle"
        >
          <GridColumn
            style={
              Object {
                "maxWidth": 600,
              }
            }
          >
            <Header
              as="h2"
              color="teal"
              textAlign="center"
            >
              SendIT Courier Services
              <Image
                as="img"
                src="logo.png"
                ui={true}
              />
            </Header>
            <Form
              as="form"
              className="form-wrapper"
              size="large"
            >
              <ButtonGroup>
                <Button
                  as="button"
                  content="Sign In"
                  icon="sign-in"
                  onClick={[Function]}
                />
                <Button
                  as="button"
                  content="Sign Up"
                  icon="user plus"
                  onClick={[Function]}
                />
              </ButtonGroup>
              <Segment
                vertical={true}
              >
                <FormField>
                  <label
                    className="label-align"
                  >
                    Email
                  </label>
                  <input
                    placeholder="Email"
                  />
                </FormField>
                <FormField>
                  <label
                    className="label-align"
                  >
                    Password
                  </label>
                  <input
                    placeholder="Password"
                    type="password"
                  />
                </FormField>
                <Button
                  as="button"
                  className="index-button"
                  color="teal"
                >
                  Login
                </Button>
              </Segment>
            </Form>
          </GridColumn>
        </Grid>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": null,
          "className": "",
          "position": "top-center",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GridColumn
            style={
              Object {
                "maxWidth": 600,
              }
            }
          >
            <Header
              as="h2"
              color="teal"
              textAlign="center"
            >
              SendIT Courier Services
              <Image
                as="img"
                src="logo.png"
                ui={true}
              />
            </Header>
            <Form
              as="form"
              className="form-wrapper"
              size="large"
            >
              <ButtonGroup>
                <Button
                  as="button"
                  content="Sign In"
                  icon="sign-in"
                  onClick={[Function]}
                />
                <Button
                  as="button"
                  content="Sign Up"
                  icon="user plus"
                  onClick={[Function]}
                />
              </ButtonGroup>
              <Segment
                vertical={true}
              >
                <FormField>
                  <label
                    className="label-align"
                  >
                    Email
                  </label>
                  <input
                    placeholder="Email"
                  />
                </FormField>
                <FormField>
                  <label
                    className="label-align"
                  >
                    Password
                  </label>
                  <input
                    placeholder="Password"
                    type="password"
                  />
                </FormField>
                <Button
                  as="button"
                  className="index-button"
                  color="teal"
                >
                  Login
                </Button>
              </Segment>
            </Form>
          </GridColumn>,
          "style": Object {
            "height": "60%",
          },
          "textAlign": "center",
          "verticalAlign": "middle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Header
                as="h2"
                color="teal"
                textAlign="center"
              >
                SendIT Courier Services
                <Image
                  as="img"
                  src="logo.png"
                  ui={true}
                />
              </Header>,
              <Form
                as="form"
                className="form-wrapper"
                size="large"
              >
                <ButtonGroup>
                  <Button
                    as="button"
                    content="Sign In"
                    icon="sign-in"
                    onClick={[Function]}
                  />
                  <Button
                    as="button"
                    content="Sign Up"
                    icon="user plus"
                    onClick={[Function]}
                  />
                </ButtonGroup>
                <Segment
                  vertical={true}
                >
                  <FormField>
                    <label
                      className="label-align"
                    >
                      Email
                    </label>
                    <input
                      placeholder="Email"
                    />
                  </FormField>
                  <FormField>
                    <label
                      className="label-align"
                    >
                      Password
                    </label>
                    <input
                      placeholder="Password"
                      type="password"
                    />
                  </FormField>
                  <Button
                    as="button"
                    className="index-button"
                    color="teal"
                  >
                    Login
                  </Button>
                </Segment>
              </Form>,
            ],
            "style": Object {
              "maxWidth": 600,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h2",
                "children": Array [
                  "SendIT Courier Services",
                  <Image
                    as="img"
                    src="logo.png"
                    ui={true}
                  />,
                ],
                "color": "teal",
                "textAlign": "center",
              },
              "ref": null,
              "rendered": Array [
                "SendIT Courier Services",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "img",
                    "src": "logo.png",
                    "ui": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "form",
                "children": Array [
                  <ButtonGroup>
                    <Button
                      as="button"
                      content="Sign In"
                      icon="sign-in"
                      onClick={[Function]}
                    />
                    <Button
                      as="button"
                      content="Sign Up"
                      icon="user plus"
                      onClick={[Function]}
                    />
                  </ButtonGroup>,
                  <Segment
                    vertical={true}
                  >
                    <FormField>
                      <label
                        className="label-align"
                      >
                        Email
                      </label>
                      <input
                        placeholder="Email"
                      />
                    </FormField>
                    <FormField>
                      <label
                        className="label-align"
                      >
                        Password
                      </label>
                      <input
                        placeholder="Password"
                        type="password"
                      />
                    </FormField>
                    <Button
                      as="button"
                      className="index-button"
                      color="teal"
                    >
                      Login
                    </Button>
                  </Segment>,
                ],
                "className": "form-wrapper",
                "size": "large",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <Button
                        as="button"
                        content="Sign In"
                        icon="sign-in"
                        onClick={[Function]}
                      />,
                      <Button
                        as="button"
                        content="Sign Up"
                        icon="user plus"
                        onClick={[Function]}
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "button",
                        "content": "Sign In",
                        "icon": "sign-in",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "button",
                        "content": "Sign Up",
                        "icon": "user plus",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <FormField>
                        <label
                          className="label-align"
                        >
                          Email
                        </label>
                        <input
                          placeholder="Email"
                        />
                      </FormField>,
                      <FormField>
                        <label
                          className="label-align"
                        >
                          Password
                        </label>
                        <input
                          placeholder="Password"
                          type="password"
                        />
                      </FormField>,
                      <Button
                        as="button"
                        className="index-button"
                        color="teal"
                      >
                        Login
                      </Button>,
                    ],
                    "vertical": true,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          <label
                            className="label-align"
                          >
                            Email
                          </label>,
                          <input
                            placeholder="Email"
                          />,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Email",
                            "className": "label-align",
                          },
                          "ref": null,
                          "rendered": "Email",
                          "type": "label",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "placeholder": "Email",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "input",
                        },
                      ],
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          <label
                            className="label-align"
                          >
                            Password
                          </label>,
                          <input
                            placeholder="Password"
                            type="password"
                          />,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Password",
                            "className": "label-align",
                          },
                          "ref": null,
                          "rendered": "Password",
                          "type": "label",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "placeholder": "Password",
                            "type": "password",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "input",
                        },
                      ],
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "button",
                        "children": "Login",
                        "className": "index-button",
                        "color": "teal",
                      },
                      "ref": null,
                      "rendered": "Login",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SemanticToastContainer
            animation={null}
            className=""
            position="top-center"
          />,
          <Grid
            style={
              Object {
                "height": "60%",
              }
            }
            textAlign="center"
            verticalAlign="middle"
          >
            <GridColumn
              style={
                Object {
                  "maxWidth": 600,
                }
              }
            >
              <Header
                as="h2"
                color="teal"
                textAlign="center"
              >
                SendIT Courier Services
                <Image
                  as="img"
                  src="logo.png"
                  ui={true}
                />
              </Header>
              <Form
                as="form"
                className="form-wrapper"
                size="large"
              >
                <ButtonGroup>
                  <Button
                    as="button"
                    content="Sign In"
                    icon="sign-in"
                    onClick={[Function]}
                  />
                  <Button
                    as="button"
                    content="Sign Up"
                    icon="user plus"
                    onClick={[Function]}
                  />
                </ButtonGroup>
                <Segment
                  vertical={true}
                >
                  <FormField>
                    <label
                      className="label-align"
                    >
                      Email
                    </label>
                    <input
                      placeholder="Email"
                    />
                  </FormField>
                  <FormField>
                    <label
                      className="label-align"
                    >
                      Password
                    </label>
                    <input
                      placeholder="Password"
                      type="password"
                    />
                  </FormField>
                  <Button
                    as="button"
                    className="index-button"
                    color="teal"
                  >
                    Login
                  </Button>
                </Segment>
              </Form>
            </GridColumn>
          </Grid>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animation": null,
            "className": "",
            "position": "top-center",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridColumn
              style={
                Object {
                  "maxWidth": 600,
                }
              }
            >
              <Header
                as="h2"
                color="teal"
                textAlign="center"
              >
                SendIT Courier Services
                <Image
                  as="img"
                  src="logo.png"
                  ui={true}
                />
              </Header>
              <Form
                as="form"
                className="form-wrapper"
                size="large"
              >
                <ButtonGroup>
                  <Button
                    as="button"
                    content="Sign In"
                    icon="sign-in"
                    onClick={[Function]}
                  />
                  <Button
                    as="button"
                    content="Sign Up"
                    icon="user plus"
                    onClick={[Function]}
                  />
                </ButtonGroup>
                <Segment
                  vertical={true}
                >
                  <FormField>
                    <label
                      className="label-align"
                    >
                      Email
                    </label>
                    <input
                      placeholder="Email"
                    />
                  </FormField>
                  <FormField>
                    <label
                      className="label-align"
                    >
                      Password
                    </label>
                    <input
                      placeholder="Password"
                      type="password"
                    />
                  </FormField>
                  <Button
                    as="button"
                    className="index-button"
                    color="teal"
                  >
                    Login
                  </Button>
                </Segment>
              </Form>
            </GridColumn>,
            "style": Object {
              "height": "60%",
            },
            "textAlign": "center",
            "verticalAlign": "middle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Header
                  as="h2"
                  color="teal"
                  textAlign="center"
                >
                  SendIT Courier Services
                  <Image
                    as="img"
                    src="logo.png"
                    ui={true}
                  />
                </Header>,
                <Form
                  as="form"
                  className="form-wrapper"
                  size="large"
                >
                  <ButtonGroup>
                    <Button
                      as="button"
                      content="Sign In"
                      icon="sign-in"
                      onClick={[Function]}
                    />
                    <Button
                      as="button"
                      content="Sign Up"
                      icon="user plus"
                      onClick={[Function]}
                    />
                  </ButtonGroup>
                  <Segment
                    vertical={true}
                  >
                    <FormField>
                      <label
                        className="label-align"
                      >
                        Email
                      </label>
                      <input
                        placeholder="Email"
                      />
                    </FormField>
                    <FormField>
                      <label
                        className="label-align"
                      >
                        Password
                      </label>
                      <input
                        placeholder="Password"
                        type="password"
                      />
                    </FormField>
                    <Button
                      as="button"
                      className="index-button"
                      color="teal"
                    >
                      Login
                    </Button>
                  </Segment>
                </Form>,
              ],
              "style": Object {
                "maxWidth": 600,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "h2",
                  "children": Array [
                    "SendIT Courier Services",
                    <Image
                      as="img"
                      src="logo.png"
                      ui={true}
                    />,
                  ],
                  "color": "teal",
                  "textAlign": "center",
                },
                "ref": null,
                "rendered": Array [
                  "SendIT Courier Services",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "img",
                      "src": "logo.png",
                      "ui": true,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "form",
                  "children": Array [
                    <ButtonGroup>
                      <Button
                        as="button"
                        content="Sign In"
                        icon="sign-in"
                        onClick={[Function]}
                      />
                      <Button
                        as="button"
                        content="Sign Up"
                        icon="user plus"
                        onClick={[Function]}
                      />
                    </ButtonGroup>,
                    <Segment
                      vertical={true}
                    >
                      <FormField>
                        <label
                          className="label-align"
                        >
                          Email
                        </label>
                        <input
                          placeholder="Email"
                        />
                      </FormField>
                      <FormField>
                        <label
                          className="label-align"
                        >
                          Password
                        </label>
                        <input
                          placeholder="Password"
                          type="password"
                        />
                      </FormField>
                      <Button
                        as="button"
                        className="index-button"
                        color="teal"
                      >
                        Login
                      </Button>
                    </Segment>,
                  ],
                  "className": "form-wrapper",
                  "size": "large",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <Button
                          as="button"
                          content="Sign In"
                          icon="sign-in"
                          onClick={[Function]}
                        />,
                        <Button
                          as="button"
                          content="Sign Up"
                          icon="user plus"
                          onClick={[Function]}
                        />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "as": "button",
                          "content": "Sign In",
                          "icon": "sign-in",
                          "onClick": [Function],
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "as": "button",
                          "content": "Sign Up",
                          "icon": "user plus",
                          "onClick": [Function],
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <FormField>
                          <label
                            className="label-align"
                          >
                            Email
                          </label>
                          <input
                            placeholder="Email"
                          />
                        </FormField>,
                        <FormField>
                          <label
                            className="label-align"
                          >
                            Password
                          </label>
                          <input
                            placeholder="Password"
                            type="password"
                          />
                        </FormField>,
                        <Button
                          as="button"
                          className="index-button"
                          color="teal"
                        >
                          Login
                        </Button>,
                      ],
                      "vertical": true,
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": Array [
                            <label
                              className="label-align"
                            >
                              Email
                            </label>,
                            <input
                              placeholder="Email"
                            />,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Email",
                              "className": "label-align",
                            },
                            "ref": null,
                            "rendered": "Email",
                            "type": "label",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "placeholder": "Email",
                            },
                            "ref": null,
                            "rendered": null,
                            "type": "input",
                          },
                        ],
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": Array [
                            <label
                              className="label-align"
                            >
                              Password
                            </label>,
                            <input
                              placeholder="Password"
                              type="password"
                            />,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Password",
                              "className": "label-align",
                            },
                            "ref": null,
                            "rendered": "Password",
                            "type": "label",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "placeholder": "Password",
                              "type": "password",
                            },
                            "ref": null,
                            "rendered": null,
                            "type": "input",
                          },
                        ],
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "as": "button",
                          "children": "Login",
                          "className": "index-button",
                          "color": "teal",
                        },
                        "ref": null,
                        "rendered": "Login",
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
